### PR TITLE
Chrisnas/docs 2799 dotnet profiler

### PR DESCRIPTION
**Pull request by chrisnas**
### What does this PR do?
Add support for .NET Profiler

### Motivation
Beta coming soon

### Preview
https://docs-staging.datadoghq.com/chrisnas/docs-2799-dotnet-profiler/tracing/profiler/enabling/dotnet/?tab=windows

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by kayayarai — APPROVED** [review]

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 1]
```diff
@@ -50,20 +50,17 @@ The profiler is shipped within Datadog tracing libraries. If you are already usi
...
-2. Run the .NET Monitoring MSI installer with administrator privileges.
-
-### Enable the profiler for your application
+3. Run the .NET Monitoring MSI installer with administrator privileges.
 
 {{< tabs >}}
 
 {{% tab "Internet Information Services (IIS)" %}}
-1. The .NET Monitoring MSI installer adds all required environment variables. There are no environment variables you need to configure.
+1. Set environment variables:
```
[the remark this answers is not in the document]
Comment (dbenamydd) [reply content/en/tracing/profiler/enabling/dotnet.md]: We still want to suggest setting env, maybe service, and version for an optimal experience, so should include that here even though the installer sets the "enable the profiler" env vars.

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 2]
```diff
@@ -75,125 +72,121 @@ Install the .NET Profiler machine-wide so that all applications on the machine c
...
-**Default**: `%ProgramData%\Datadog-APM\logs\`
 
-`DD_PROFILING_ENABLED`
-: If set to `false`, disables the .NET Profiler. <br>
-**Default**: `true`
+## Configuration
 
+You can configure the profiler using the following environment variables. Restart the application after any of these settings is changed.
+
+| Environment variable                             | Type          | Description                                                                                      |
```
[the remark this answers is not in the document]
Comment (dbenamydd) [reply content/en/tracing/profiler/enabling/dotnet.md:166]: Gotcha. Which tracer doc are you comparing with? I think there are a couple.

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 3]
```diff
@@ -50,20 +50,17 @@ The profiler is shipped within Datadog tracing libraries. If you are already usi
   <strong>Note:</strong> Datadog's.NET Tracer and Profiler rely on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.
 </div>
 
-### Install the profiler
+1. If you are already using Datadog, upgrade your agent to version [7.20.2][2]+ or [6.20.2][3]+.
 
-Install the .NET Profiler machine-wide so that all applications on the machine can be profiled:
+2. Download the [.NET Monitoring MSI installer][1] version TODO+. Continuous Profiler only supports 64-bit, with 32-bit support coming soon.
```
[the remark this answers is not in the document]
Comment (dbenamydd) [reply content/en/tracing/profiler/enabling/dotnet.md]: We do this in other language pages so if, in the future, an apm user wants to enable profiling, if they're already running a new enough version, they can skip this step. There are pros and cons to this but if we want to get rid of it should do it for all languages.

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 4]
```diff
@@ -75,125 +72,121 @@ Install the .NET Profiler machine-wide so that all applications on the machine c
...
+{{% tab "Standalone applications" %}}
+To automatically profile a non-service application, such as console, ASP.NET (Core), Windows Forms, or WPF, you'll need to set some environment variables before starting it.
 
-### Optional configuration
+If the environment variables are set for the current user, _all_ .NET applications will be profiled. Instead, we recomment setting them in a batch file that also starts the application and running that. For example:
 
-The following configuration variables are available for customizing .NET Profiler behavior:
+ - For .NET Core and .NET 5+:
+ ```
+ TODO show example batch file
```
[the remark this answers is not in the document]
Comment (dbenamydd) [reply content/en/tracing/profiler/enabling/dotnet.md]: Thanks!

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+64-bit applications running on:
+- .NET Framework 4.6.1+
+- .NET Core 2.1, 3.1
+- .NET 5
+- .NET 6
+
+## Installation
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> Datadog's.NET Tracer and Profiler rely on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:47]:
```suggestion
  <strong>Note:</strong> Datadog's .NET Tracer and Profiler rely on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.
```

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+   [string[]] $v = @(
+       "COR_ENABLE_PROFILING=1",
+       "COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}",
+       "DD_ENV=prod",
+       "DD_VERSION=1.2.3"
+   )
+   Set-ItemProperty HKLM:SYSTEM\CurrentControlSet\Services\<SERVICE NAME> -Name Environment -Value $v
+   ```
+
+2. A minute or two after starting your application, your profiles will show up on the [Datadog APM > Profiler page][1].
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:126]:
```suggestion
2. A minute or two after you start your application, your profiles appear on the [Datadog APM > Profiler page][1].
```

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+[1]: https://app.datadoghq.com/profiling
+{{% /tab %}}
+
+{{% tab "Standalone applications" %}}
+1. To automatically profile a non-service application, such as console, ASP.NET (Core), Windows Forms, or WPF, some environment variables must be set before starting it.
+
+   If the environment variables are set for the current user, _all_ .NET applications will be profiled. Instead, we recommend setting them in a batch file that also starts the application and running that.
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:134]:
```suggestion
   If the environment variables are set for the current user, _all_ .NET applications are profiled. Instead, set them in a batch file that also starts the application, and run the batch file.
```

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+   ```cmd
+   SET COR_ENABLE_PROFILING=1
+   SET COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+   SET DD_ENV=prod
+   SET DD_VERSION=1.2.3
+
+   REM start the application here
+   ```
+
+2. A minute or two after starting your application, your profiles will show up on the [Datadog APM > Profiler page][1].
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:156]:
```suggestion
2. A minute or two after you start your application, your profiles appear on the [Datadog APM > Profiler page][1].
```

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+| `DD_ENV`                   | String        | The [environment][4] name, for example, `production`. |
+| `DD_SERVICE`               | String        | The [service][4] name, for example, `web-backend`. If this is not specified, the .NET Profiler tries to determine the service name automatically from the application name (process entry assembly or process name).    |
+| `DD_VERSION`               | String        | The [version][4] of your service.  |
+| `DD_TAGS`                  | String        | Tags to apply to an uploaded profile. Must be a list of `<key>:<value>` separated by commas such as: `layer:api,team:intake`.   |
+| `DD_AGENT_HOST`            | String        | Sets the host where profiles are sent (the host running the Agent). Can be a hostname or an IP address. Ignored if `DD_TRACE_AGENT_URL` is set. Defaults to `localhost`.  |
+| `DD_TRACE_AGENT_PORT`      | String        | Sets the port where profiles are sent (the port where the Agent is listening for connections). Ignored if `DD_TRACE_AGENT_URL` is set. Defaults to`8126`..  |
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:173]:
```suggestion
| `DD_TRACE_AGENT_PORT`      | String        | Sets the port where profiles are sent (the port where the Agent is listening for connections). Ignored if `DD_TRACE_AGENT_URL` is set. Defaults to`8126`.  |
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+## Installation
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> Datadog's .NET Tracer and Profiler rely on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.
+</div>
+
+1. If you are already using Datadog, upgrade your agent to version [7.20.2][1]+ or [6.20.2][2]+.
+
+2. The profiler and tracer are shipped together. Install or upgrade to version TODO+ if needed, using the [.NET Monitoring MSI installer][3]. Continuous Profiler only supports 64-bit Windows, with 32-bit support coming soon, so you'll want the file like `datadog-dotnet-apm-2.1.0-x64.msi`.
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:52]: Commenting here to make sure we don't merge without fixing this TODO.

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+## Installation
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> Datadog's .NET Tracer and Profiler rely on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.
+</div>
+
+1. If you are already using Datadog, upgrade your agent to version [7.20.2][1]+ or [6.20.2][2]+.
+
+2. The profiler ships together with the tracer. Install or upgrade to the last beta version if needed, using the [.NET Monitoring MSI installer][3]. Continuous Profiler only supports 64-bit Windows, with 32-bit support coming soon, so you need the file like `datadog-dotnet-apm-2.1.0-x64-profiler-beta.msi`.
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:54]:
```suggestion
2. The profiler ships together with the tracer. Install or upgrade to the last beta version if needed, using the [.NET Monitoring MSI installer][3]. Continuous Profiler only supports 64-bit Windows, so you need the file like `datadog-dotnet-apm-2.1.0-x64-profiler-beta.msi`.
```

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+
+   Run the .NET Monitoring MSI installer with administrator privileges.
+
+**Note**: Profiling requires the .msi version 2.1
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:58]:
```suggestion
**Note**: Profiling requires the `.msi` version 2.1.
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: content/en/tracing/profiler/enabling/dotnet.md [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+   Set-ItemProperty HKLM:SYSTEM\CurrentControlSet\Services\<SERVICE NAME> -Name Environment -Value $v
+   ```
+
+2. A minute or two after you start your application, your profiles appear on the [Datadog APM > Profiler page][1].
+
+[1]: https://app.datadoghq.com/profiling
+{{% /tab %}}
+
+{{% tab "Standalone applications" %}}
+1. Set needed environment variables to configure and enable Profiler for a non-service application, such as console, ASP.NET (Core), Windows Forms, or WPF. You can do this machine-wide, user-wide, or just for your service. We recommend doing this in a batch file that also starts the application, and running your application using the batch file.
```
Comment (kayayarai) [inline content/en/tracing/profiler/enabling/dotnet.md:190]:
```suggestion
1. Set needed environment variables to configure and enable Profiler for a non-service application, such as console, ASP.NET (Core), Windows Forms, or WPF. You can do this machine-wide, user-wide, or just for your service. The recommended way to do this is in a batch file that also starts the application, and to run your application using the batch file.
```

**Review by kayayarai — APPROVED** [review]

File: content/en/tracing/profiler/_index.md [hunk 14]
```diff
@@ -52,7 +52,7 @@ Profiling your service to visualize all your stack traces in one place takes jus
 
 {{< partial name="profiling/profiling-languages.html" >}}
 
-To get notified when a private beta is available for the **PHP** or **.NET** Profiler, complete the [Continuous Profiler - Product Access Request form][1].
+To get notified when a private beta is available for the **PHP** Profiler, complete the [Continuous Profiler - Product Access Request form][1].
```
Comment (kayayarai) [inline content/en/tracing/profiler/_index.md:55]:
```suggestion

```

**Pull request merged**